### ISSUE TITLE
Avoid plugin build warnings in addTargetDependencies on Linux 

### ISSUE THREAD

**Issue by ktoso** [apple/swift-docc-plugin#83]
We're seeing the following warnings when using the plugin on Linux:

```
13:02:05 /tmp/.check-api_EHIjZG/repo/.build/x86_64-unknown-linux-gnu/apidiff/d16e216341cd13c4f72bf53164dd39acbf14ebbd-checkout/.build/checkouts/swift-docc-plugin/Plugins/Swift-DocC Preview/Symbolic Links/SharedPackagePluginExtensions/PackageExtensions.swift:31:17: warning: switch covers known cases, but 'TargetDependency' may have additional unknown values
13:02:05                 switch dependency {
13:02:05                 ^
13:02:05 /tmp/.check-api_EHIjZG/repo/.build/x86_64-unknown-linux-gnu/apidiff/d16e216341cd13c4f72bf53164dd39acbf14ebbd-checkout/.build/checkouts/swift-docc-plugin/Plugins/Swift-DocC Preview/Symbolic Links/SharedPackagePluginExtensions/PackageExtensions.swift:31:17: note: handle unknown values using "@unknown default"
13:02:05                 switch dependency {
13:02:05                 ^
```

And the respective source is https://github.com/apple/swift-docc-plugin/blob/main/Plugins/SharedPackagePluginExtensions/PackageExtensions.swift#L36-L40

```
                #if os(macOS) || os(iOS) || os(tvOS) || os(watchOS)
                @unknown default:
                    return
                #endif
```

Why is this handling only apple platforms - can this be removed so we avoid warning on Linux builds?

**Comment by diegolavalledev** [comment]
Looks like it was addressed by #67

**Comment by ktoso** [comment]
Seems so!
